File: solutions/Question 2114/Solution5.py
# Problem: 2114. Maximum Number of Words Found in Sentences
# Difficulty: Easy
# Topics: Array, String
# URL: https://leetcode.com/problems/maximum-number-of-words-found-in-sentences/
# Runtime: 0 ms
# Memory: 17.9 MB
# Solved via CodePath Auto-Committer

class Solution:
    def mostWordsFound(self, sentences: List[str]) -> int:
        # a = 0
        # for b in sentences:
            # x = len(b.split(" "))
            # if x > a:
                # a = x
        # return a
        return max(len(sentence.split()) for sentence in sentences)

File: solutions/Question 3697/Solution5.cpp
/*
 * Problem: 3697. Compute Decimal Representation
 * Difficulty: Easy
 * Topics: Array, Math
 * URL: https://leetcode.com/problems/compute-decimal-representation/
 * Runtime: 3 ms
 * Memory: 9.7 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    vector<int> decimalRepresentation(int n) {
        long long num = 1;
        vector<int> ans;
        while(n>0){
            int x = n%10;
            x *= num;
            num *= 10; // giving error if num is int;
            if(x != 0)
                ans.push_back(x);
            n = n/10;
        }
        reverse(ans.begin(),ans.end()); // we can also just sort reverse;
        // sort(ans.rbegin(),ans.rend()); // but after this runtine is 3ms not 0 
        return ans;  
        
    }
};